base/intfuncs.jl

```diff
@@ -171,18 +172,23 @@ function power_by_squaring(x_, p::Integer)
     end
     t = trailing_zeros(p) + 1
     p >>= t
+    o = false
     while (t -= 1) > 0
-        x *= x
+        x, o′ = mul_with_overflow(x, x)
+        o |= o′
     end
     y = x
     while p > 0
         t = trailing_zeros(p) + 1
         p >>= t
         while (t -= 1) >= 0
-            x *= x
+            x, o′ = mul_with_overflow(x, x)
+            o |= o′
         end
-        y *= x
+        y, o′ = mul_with_overflow(y, x)
+        o |= o′
     end
+    o && throw(OverflowError)
```

Comment:
yes, already fixed locally. I just figured I'd start the policy discussion with some code and numbers ;).